mailbox/event/json/src/main/scala/org/apache/james/event/json/MetaDataDTO.scala

```diff
@@ -0,0 +1,89 @@
+/** **************************************************************
+  * Licensed to the Apache Software Foundation (ASF) under one   *
+  * or more contributor license agreements.  See the NOTICE file *
+  * distributed with this work for additional information        *
+  * regarding copyright ownership.  The ASF licenses this file   *
+  * to you under the Apache License, Version 2.0 (the            *
+  * "License"); you may not use this file except in compliance   *
+  * with the License.  You may obtain a copy of the License at   *
+  * *
+  * http://www.apache.org/licenses/LICENSE-2.0                 *
+  * *
+  * Unless required by applicable law or agreed to in writing,   *
+  * software distributed under the License is distributed on an  *
+  * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+  * KIND, either express or implied.  See the License for the    *
+  * specific language governing permissions and limitations      *
+  * under the License.                                           *
+  * ***************************************************************/
+
+package org.apache.james.event.json
+
+import java.time.Instant
+import java.util.Date
+
+import javax.mail.{Flags => JavaMailFlags}
+import org.apache.james.mailbox.model.{MessageId, MessageMetaData => JavaMessageMetaData}
+import org.apache.james.mailbox.{FlagsBuilder, MessageUid}
+
+object MetaDataDTO {
+
+  object MessageMetaData {
+    def fromJava(javaMessageMetaData: JavaMessageMetaData): MessageMetaData = MetaDataDTO.MessageMetaData(
+      javaMessageMetaData.getUid,
+      javaMessageMetaData.getModSeq,
+      javaMessageMetaData.getFlags,
+      javaMessageMetaData.getSize,
+      javaMessageMetaData.getInternalDate.toInstant,
+      javaMessageMetaData.getMessageId)
+  }
+
+  object Flags {
+    val ANSWERED = "\\Answered"
+    val DELETED = "\\Deleted"
+    val DRAFT = "\\Draft"
+    val FLAGGED = "\\Flagged"
+    val RECENT = "\\Recent"
+    val SEEN = "\\Seen"
+    val ALL_SYSTEM_FLAGS = List(ANSWERED, DELETED, DRAFT, FLAGGED, RECENT, SEEN)
```

Comment:
I think that the serialization layer is not responsible for knowing the system flags list.
We should keep this kind of logic in the Java code.

Reply:
There is a class `MessageFlags` contains all those flags in string, but IDK does it make sense to import a class from `org.apache.james.imap.api.message`?

Reply:
What I mean is that it probably belongs to the flagsbuilder, you should avoid any logic that is not simple translation between java and scala object.
If you need some logic, please put it in the Java objects

Reply:
Agree we could have a static method in `mailbox/api` responsible of converting a Flags into a List of Strings

Reply:
updated https://github.com/linagora/james-project/pull/2039/commits/cf50903065e8ac72422db38dd12efe24e6bf5b6b